doc-src/install.rst

```diff
@@ -1,25 +1,330 @@
 Installation
 ************
 
-This section describes how to build and install Charliecloud. For some
-distributions, this can be done using your package manager; otherwise, both
-normal users and admins can build and install it manually.
+This section describes what you need to install Charliecloud and how to do so.
+
+Note that installing and using Charliecloud requires no privilege, provided
+that user namespaces have been enabled in the kernel.
 
 .. contents::
    :depth: 2
    :local:
 
 
+Build and install from source
+=============================
+
+The tarballs we provide include the build system (:code:`configure`, etc.) and
+pre-built documentation. Thus, build and install is a standard::
+
+  $ ./configure
+  $ make
+  $ sudo make install
+
+If you don't have sudo, you can:
+
+  * Run Charliecloud directly from the build directory; add
+    :code:`$BUILD_DIR/bin` to your :code:`$PATH` and you are good to go,
+    without :code:`make install`.
+
+  * Install in a prefix you have write access to, e.g. in your home directory
+    with :code:`./configure --prefix=~`.
+
+:code:`configure` will provide a detailed report on what will be built and
+installed along with what dependencies are present and missing.
+
+.. note::
+
+   A Git checkout (or tarball after :code:`make maintainer-clean`) will not
+   have :code:`configure` or pre-built documentation. To bootstrap, you need
+   GNU Autotools installed. Run the helper script
+   :code:`configure-bootstrap.sh`.
```

Comment:
The idiomatic term for this bootstrap script is `autogen.sh`. I think we should rename it to reduce friction with people who already use autotools based projects.